iceth-API.md

```diff
@@ -0,0 +1,107 @@
+# IC Eth API
+
+## Overview
+
+*Service, JSON API service:* A Web2 service such as [Infura](https://www.infura.io/), [Gateway.fm](https://gateway.fm/), or [CloudFlare](https://www.cloudflare.com/en-gb/web3/) that offers access to the Ethereum JSON RPC API through HTTP. Note that also other EVM-compatible chains may be covered by such JSON RPC API.
+
+*Provider:* A provider is registered in the canister and allows for connecting to a specific JSON API service in the Web2 world. Each chain id for a particular service requires a different provider and typically requires a different API key. Multiple providers can be registered for a service / chain id combination.
+
+*Chain id*: The chain id determines the Ethereum / EVM network to connect to.
+
+## Data Types
+
+## Methods
+
+### register_provider
+
+Register a new *provider* for a Web2-based *service*.
+
+    type RegisterProvider = record {
+        chain_id: nat64;
+        service_url: text;
+        api_key: text;
+        cycles_per_call: nat64;
+        cycles_per_message_byte: nat64;
+    };
+
+    register_provider: (RegisterProvider) -> ();
+
+The `RegisterProvider` record defines the details about the service to register, including the API key for the service.
+* `chain_id`: The id of the Ethereum chain this provider allows to connect to. The ids refer to the chain ids as defined for EVM-compatible blockchains, see, e.g., [ChainList](https://chainlist.org/?testnets=true).
+* `service_url`: The URLs of the Web2 service provider that is used by the canister when using this provider.
+* `api_key`: The API key for authorizing requests to this service provider.
```

Comment:
We should add that the API is private to the registerer and the canister and is not return in by the get_providers() call.  The complete URL used to access the service is constructed by concatenating the service_url and the api_key (without a seperator) e.g. "https://cloudflare-eth.com" and "/my-api-key").

Reply:
Good points!